aws/src/integration/java/org/apache/iceberg/aws/TestAssumeRoleAwsClientFactory.java

```diff
@@ -58,7 +60,7 @@ public void before() {
     roleName = UUID.randomUUID().toString();
     iam = IamClient.builder()
         .region(Region.AWS_GLOBAL)
-        .httpClient(UrlConnectionHttpClient.create())
+        .httpClientBuilder(HTTP_CLIENT_BUILDER_DEFAULT)
```

Comment:
Based on your link:

> When there is no desire to share HTTP clients across multiple service clients, the client builder is the preferred way to customize the HTTP client as it benefits from service-specific default configuration.

So we should initialize a new HTTP client for every client, they should not share the HTTP client.

Reply:
To my understanding we are just sharing the builder in this way and we are not creating any http clients just yet. The AWS client will use this builder to create a separate http client and will manage its life cycle. 

Am I missing something?

Reply:
I think there might be some race conditions, especially if this is shared globally. Based on the documentation, services will inject necessary configurations to the HTTP client builder, so at the same time, there might be 2 processes creating different clients, but if they share the same builder that means the HTTP client produced will be e mix of the 2 service configs. This is something we should better avoid. Given the fact that now each client has its own HTTP client, reusing the builder is not really saving any resource since the builder is ephemeral, so I think it is safer to use individual builders.